### PR TITLE
Rename module to multicluster-gateway-controller

### PR DESCRIPTION
**Pull request by mikenairn**
Rename module multi-cluster-traffic-controller -> multicluster-gateway-controller

### CODE REVIEW

File: docs/milestone2/milestone2_application.yaml [hunk 1]
```diff
@@ -1,39 +1,39 @@
 apiVersion: v1
```
Comment (david-martin) [inline docs/milestone2/milestone2_application.yaml:1]: These milestone 2 files can be removed on main and live on on the milestone 2 tag.
The steps won't work on main regardless of the mctc/mgc change. (relies on syncer pre-ocm changes)

I'll PR that separate after merge of this if it's simpler

**Review by david-martin — COMMENTED** [review]
/approve

File: docs/milestone2/milestone2_application.yaml (hunk 1 above)
Comment (mikenairn) [reply docs/milestone2/milestone2_application.yaml:1]: There seems to be quite a few things that need to be removed from this repo, syncer, RLP controller, outdated docs etc ... I'd rather just merge this, and do other follow up PRs for the rest if possible.

Comment (mikenairn) [reply docs/milestone2/milestone2_application.yaml:1]: Follow up to remove RLP and milestone 2 docs https://github.com/Kuadrant/multicluster-gateway-controller/pull/223

**Pull request merged**